kratos/processes/replace_elements_and_condition_process.cpp

```diff
@@ -19,77 +19,83 @@
+        auto it_entity = rEntityContainer.begin() + i;
 
-        auto p_element = rReferenceElement.Create(it_elem->Id(), it_elem->pGetGeometry(), it_elem->pGetProperties());
+        auto p_new_entity = rReferenceEntity.Create(it_entity->Id(), it_entity->pGetGeometry(), it_entity->pGetProperties());
 
-        // Deep copy elemental data and flags
-        p_element->Data() = it_elem->Data();
-        p_element->Set(Flags(*it_elem));
+        // Deep copy data and flags
+        p_new_entity->Data() = it_entity->Data();
+        p_new_entity->Set(Flags(*it_entity));
 
-        (*it_elem.base()) = p_element;
+        (*it_entity.base()) = p_new_entity;
     }
+}
 
-    #pragma omp parallel for
-    for(int i=0; i< static_cast<int>(r_root_model_part.Conditions().size()); i++) {
-        auto it_cond = r_root_model_part.ConditionsBegin() + i;
+void UpdateSubModelPart(
+    ModelPart& rModelPart,
+    ModelPart& rRootModelPart,
+    const bool UpdateElements,
+    const bool UpdateConditions
+    )
+{
+    // Change the model part itself
+    if (UpdateElements) {
+        #pragma omp parallel for
+        for(int i=0; i< static_cast<int>(rModelPart.Elements().size()); i++) {
+            auto it_elem = rModelPart.ElementsBegin() + i;
 
-        auto p_condition = rReferenceCondition.Create(it_cond->Id(), it_cond->pGetGeometry(), it_cond->pGetProperties());
+            (*it_elem.base()) = rRootModelPart.Elements()(it_elem->Id());
+        }
+    }
 
-        // Deep copy elemental data and flags
-        p_condition->Data() = it_cond->Data();
-        p_condition->Set(Flags(*it_cond));
+    if (UpdateConditions) {
+        #pragma omp parallel for
+        for(int i=0; i< static_cast<int>(rModelPart.Conditions().size()); i++) {
+            auto it_cond = rModelPart.ConditionsBegin() + i;
 
-        (*it_cond.base()) = p_condition;
+            (*it_cond.base()) = rRootModelPart.Conditions()(it_cond->Id());
+        }
     }
 
     // Change the submodelparts
-    for (auto& i_sub_model_part : r_root_model_part.SubModelParts()) {
-        UpdateSubModelPart( i_sub_model_part, r_root_model_part );
+    for (auto& r_sub_model_part : rModelPart.SubModelParts()) {
+        UpdateSubModelPart(r_sub_model_part, rRootModelPart, UpdateElements, UpdateConditions);
     }
 }
 
+}
+
 /***********************************************************************************/
 /***********************************************************************************/
 
-void ReplaceElementsAndConditionsProcess::UpdateSubModelPart(
-    ModelPart& rModelPart,
-    ModelPart& rRootModelPart
-    )
+void ReplaceElementsAndConditionsProcess::Execute()
 {
-    // Change the model part itself
-    #pragma omp parallel for
-    for(int i=0; i< static_cast<int>(rModelPart.Elements().size()); i++) {
-        auto it_elem = rModelPart.ElementsBegin() + i;
+    ModelPart& r_root_model_part = mrModelPart.GetRootModelPart();
```

Comment:
I see that you decided to keep using the RootModelPart. I guess the reason for this is what we already discussed that we still need to find the way of replacing the elements (and/or conditions) from a submodelpart and making sure that the replacement also takes place in all of the existing submodelparts, am I right? I mean, I just wanted to leave here the idea that it would be nice to have the feature of being able to replace the elements (and/or conditions) of just one modelpart. We just need to come up with an idea how to do it :-)

Reply:
yep as we discussed this is a more severe change that I as of now would not know how to do

As you said the main question is how the elements should be updated that also exist in other submodelparts 
Maybe it can be done using the `AssignUniqueModelPartCollectionTagUtility` but I am not sure

Reply:
I did not know about the AssignUniqueModelPartCollectionTagUtility, Ill have a look.